doc/kci_testsuite.md

```diff
@@ -0,0 +1,289 @@
+
+-  If you want to make minor modification. Then check below options:
+     - If you need to install specific Debian package, have look at `extra_packages` option in `rootfs-configs.yaml` file.
+     - To install specific test suite built from source, pep3 etc have a look at `script` option in `rootfs-configs.yml` file. It allows to you
+        run prepare rootfs image in a desired manner.
+     - To install specific files or helper scripts on specific location, have a look at `overlay` option in `rootfs-configs.yml` file.
+
+
+### How to add test suite to a rootfs image
+
+1. Let's understand how test suite is added to a rootfs image. For this, we will explore `buster-v4l2` rootfs image.
+
+2. Take a look at `rootfs-configs.yaml` for `buster-v4l2`. It should display following config entries.
+
+
+```
+  buster-v4l2:
+    rootfs_type: debos
+    debian_release: buster
+    arch_list:
+      - amd64
+      - arm64
+      - armhf
+    extra_packages_remove:
+      - bash
+      - e2fslibs
+      - e2fsprogs
+    script: "scripts/buster-v4l2.sh"
+    test_overlay: "overlays/v4l2"
+```
+
+
+3. We will look into last three config entries namely, `extra_packages_remove`, `script` and `test_overlay`.
+
+   `extra_packages_remove` as name suggests it will remove the listed packages from final rootfs image. In this case,
+    packages like bash, e2fslibs and e2fsprogs are deleted from the image.
+
+4. Understand that `script` path is relative to `kci_rootfs build --data-path` value. By default its `jenkins/debian/debos/script`
+   Lets have quick look at its current contents:
+
+```
+$ ls jenkins/debian/debos/scripts/
+
+buster-cros-ec-tests.sh  buster-igt.sh  buster-v4l2.sh  create_initrd_ramdisk.sh  create_manifest.py  crush.sh  nothing.sh  setup-networking.sh  stretch-ltp.sh  strip.sh
+```
+
+   Lets focus on `buster-v4l2.sh` for this example. If you peek into the script you will find that script performs actions like installing packages via apt-get, setting 
+   up v4l2 build directories and building and installing the v4l2 binaries. Its just a normal bash script not tied to debos or kernelci. 
+
+
+   if you like to run the `buster-v4l2.sh` script and verify manually, simple copy the script to `debian:buster` docker image and execute it.
+
+5. `test_overlay` option is used to create specific directory layout inside the rootfs image. In our case, 
+
+```
+$ tree jenkins/debian/debos/overlays/v4l2/
+jenkins/debian/debos/overlays/v4l2/
+└── usr
+    └── bin
+        └── v4l2-parser.sh
+
+```
+   `/usr/bin/v4l2-parser.sh` will be available on `buster-v4l2` rootfs image. These overlay scripts normally ran from LAVA job to parse the results.
+    [Here](https://lava.collabora.co.uk/scheduler/job/2446126/definition#defline120) is a sample LAVA job using `v4l2-parser.sh`.
+
+
+6. Hopefully now you have some basic idea about how test-suite are added to rootfs and how its used in LAVA environment.
```

Comment:
```suggestion
6. Hopefully now you should have some basic idea about how to add a test suite to a rootfs image and how it can be used in a LAVA environment.
```